iosApp/iosApp/ContentView.swift

```diff
@@ -286,16 +286,18 @@ struct ContentView: View {
                             viewportProvider: viewportProvider
                         )
                         .toolbar(.hidden, for: .tabBar)
-                        .onAppear {
-                            let filtered = stopFilter != nil
-                            screenTracker.track(
-                                screen: filtered ? .stopDetailsFiltered : .stopDetailsUnfiltered
-                            )
-                        }
                     }
                     // Set id per stop so that transitioning from one stop to another is handled by removing
                     // the existing stop view & creating a new one
                     .id(stopId)
+                    .onChange(of: stopId) { nextStopId in stopDetailsVM.handleStopChange(nextStopId) }
+                    .onAppear {
+                        stopDetailsVM.handleStopAppear(stopId)
+                        screenTracker.track(
+                            screen: stopFilter != nil ? .stopDetailsFiltered : .stopDetailsUnfiltered
+                        )
+                    }
+                    .onDisappear { stopDetailsVM.leaveStopPredictions() }
```

Comment:
See my notes in the PR description, `StopDetailsPage` gets torn down and recreated every time the stop ID or stop filter changes, which was resulting in some issues. The `onChange` handles disconnecting and reconnecting from the channel, so we only actually want `onAppear` or `onDisappear` to be called when the page is changed entirely, not just changed to a different stop/filter.

Reply:
Oh right, thank you, I wasn't connecting the dots on how this fit in with that. 

One other non-blocking thought that comes to mind based on that - it could help to clarify the behavior to add a new component to the stop details hierarchy to clarify w/ naming and/or comments that there is this stable stop details container that is unchanging and presented once, and that the inner piece is more volatile. Something like `StopDetailsPageContainer` (stable) -> `StopDetailsPage` -> `StopDetailsView`.

Page vs Container doesn't feel meaningfully distinctive as a name, but I think clarifying the difference between these layers could help with understandability down the line.

Reply:
Yeah I like that idea, I can try to fit that into one of the follow up PRs!